### ISSUE TITLE
Problem: wrong logic in modifier

### ISSUE THREAD

**Issue by CTAPCKPIM** [holdex/payment-streaming#95]
Code:
```JavaScript
    ///@dev check if it's terminated or not
    modifier notTerminated {
        if (streamInstances[msg.sender].isTerminated)
            revert AlreadyTerminated();
        _;
    }

    /**
     * @dev terminate the stream instance
     * @param _payee payee's address
     */
    function terminate(address _payee) external onlyPayer notTerminated {
        uint256 terminatedAt = block.timestamp;
        OpenStream storage streamInstance = streamInstances[_payee];
        if (!isPayee[_payee]) revert NotPayee();
        if (streamInstance.terminatedAt != 0) revert AlreadyTerminatedOrTerminating();
        /// Terminate in cliff period
        if (streamInstance.createdAt + streamInstance.cliffPeriod >= block.timestamp)
            isPayee[_payee] = false;
        streamInstance.isTerminated = true;
        streamInstance.terminatedAt = terminatedAt;

        emit StreamTerminated(_payee);
    }
```
The **modifier** is slightly erroneously spelled out:

1) When calling the function, we check the state of the variable `true/false` for the `payee` in the `modifier`, and if it is `false`, we **close** the stream by the called function

    In turn, when I rewrote the tests, I noticed that not the address of the `payee` for checking `true/false` is transmitted to the `modifier`, but the address of the `payer (msg.sender)`

- In this situation, it turns out that absolutely always the check in the modifier will return `false` since it will **default to an empty value** for any new address that is not set to `true`

2) **Another question** is whether we actually need two identical `reverts`, the difference is in what they are created through

- ` revert AlreadyTerminated();` - **modifier** 
- `revert AlreadyTerminatedOrTerminating();` - **in the function**

**Comment by CTAPCKPIM** [comment]
@zolotokrylin 
What do you think?

**Comment by georgeciubotaru** [comment]
@CTAPCKPIM Please analyze the current PR #94 I think this problem is contain there.

**Comment by zolotokrylin** [comment]
@CTAPCKPIM which PR has resolved this issue?

**Comment by CTAPCKPIM** [comment]
@zolotokrylin 
Sorry that I closed the RP itself, it was solved by #94
